web/client/utils/openlayers/DrawSupportUtils.js

```diff
@@ -12,12 +12,15 @@ import { reproject } from '../CoordinatesUtils';
 import { getCenter } from 'ol/extent';
 import { Circle } from 'ol/geom';
 
-const calculateRadius = (center, coordinates) => {
-    return isArray(coordinates) && isArray(coordinates[0]) && isArray(coordinates[0][0]) ? Math.sqrt(Math.pow(center[0] - coordinates[0][0][0], 2) + Math.pow(center[1] - coordinates[0][0][1], 2)) : 100;
+const calculateRadius = (center, coordinates, mapCrs, featureCrs) => {
+    if (isArray(coordinates) && isArray(coordinates[0]) && isArray(coordinates[0][0])) {
+        const point = reproject(coordinates[0][0], featureCrs, mapCrs);
+        return Math.sqrt(Math.pow(center[0] - point.x, 2) + Math.pow(center[1] - point.y, 2));
+    }
+    return 100;
 };
 
-export const transformPolygonToCircle = (feature, mapCrs) => {
-
+export const transformPolygonToCircle = (feature, mapCrs, featureCrs = mapCrs) => {
```

Comment:
What manual tests have you done related to this change?

For extra security I would check also circle areas in the spatial filter section of QueryPanel

Reply:
Not sure what you mean by checking circle areas in the spatial filter section. I could only find Filter Layer option where in the "Region of Interest" you can draw circle areas, but transfromPolygonToCircle is not called there so I don't quite understand how it is relevant to the issue. Could you please elaborate a little further on what you meant? I've tested the changes by adding Circle annotations in different projections then switching the CRS. Circle annotations now don't change their sizes.

Reply:
Nevermind. While I was thinking to draw functionalities of circles in mapstore it came to my mind the one present in query panel (ROI). 
Since there is another issue https://github.com/geosolutions-it/MapStore2/issues/3922 already opened, you can ignore my previous comment. 
Anyway that issue could be related to your changes (query panel uses drawSupport to start drawing etc) and you can keep this in mind for the future